### PR TITLE
NAS-118811 / 22.12 / Phase out kubernetes asyncio library

### PR DESCRIPTION
**Pull request by sonicaj**

### CODE REVIEW

File: src/middlewared/middlewared/plugins/kubernetes_linux/k8s/watch.py
Comment (themylogin) [reply src/middlewared/middlewared/plugins/kubernetes_linux/k8s/watch.py:26]: You allow binary data that fails to be decoded from utf-8 to be returned as just binary data, is this intentional?

Comment (sonicaj) [reply src/middlewared/middlewared/plugins/kubernetes_linux/k8s/watch.py:27]: It was just an extra fail safe but of course it would break at consumer side then.
I have modified it now and going to raise an exception to fail hard if this happens.

**Review by themylogin — APPROVED** [review]
Thank you!

**Review by yocalebo — APPROVED** [review]
Looks good from my perspective but let's wait for Rehan's review before merging as well. Thanks!

File: src/middlewared/middlewared/plugins/kubernetes_linux/k8s/client.py
Comment (Qubad786) [inline src/middlewared/middlewared/plugins/kubernetes_linux/k8s/client.py:46]: Should be `resp`

File: src/middlewared/middlewared/plugins/kubernetes_linux/k8s/client.py
Comment (Qubad786) [inline src/middlewared/middlewared/plugins/kubernetes_linux/k8s/client.py:69]: I guess it would be beneficial to have at least doc string here

File: src/middlewared/middlewared/plugins/kubernetes_linux/k8s/client.py
Comment (Qubad786) [inline src/middlewared/middlewared/plugins/kubernetes_linux/k8s/client.py:108]: Maybe have an enum for modes?

File: src/middlewared/middlewared/plugins/kubernetes_linux/k8s/watch.py
Comment (Qubad786) [inline src/middlewared/middlewared/plugins/kubernetes_linux/k8s/watch.py:35]: Is this being used anywhere?

File: src/middlewared/middlewared/plugins/kubernetes_linux/k8s/core_api.py
Comment (Qubad786) [inline src/middlewared/middlewared/plugins/kubernetes_linux/k8s/core_api.py:90]: `cls.STREAM_RESPONSE_TIMEOUT * 60` ?

**Review by Qubad786 — COMMENTED** [review]
I think overall it looks great, just small nits can be addressed.

**Review by Qubad786 — APPROVED** [review]

**Pull request merged**
